torch/_dynamo/stream.py

```diff
@@ -0,0 +1,98 @@
+import torch
+import torch.cuda
+import warnings
+
+
+def __singleton(cls):
+    instance = {}
+
+    def get_instance(*args, **kwargs):
+        if cls not in instance:
+            instance[cls] = cls(*args, **kwargs)
+        return instance[cls]
+    return get_instance
```

Comment:
Why is instance a dict?   Won't it only have one member?

If args/kwargs change, the second version will be silently ignored.  This seems error prone.

Reply:
Thank you for comment. Here `__singleton` is only used for  `StreamMethodContainer`, so it has one member. Maybe this decorator can be used in somewhere developer needs to implement a singleton class.

When args/kwargs changed, only the initial created instance is returned, here needs a warning, otherwise there will be a silent ignore.

Reply:
This decorator has been removed.